### PR TITLE
Use consistent potential temperature for model-obs comparison

### PR DESCRIPTION
**Pull request by YanchunHe**
**Background information:**
* the model climatologies of BLOM are compared with the World Ocean Atlas (version 2013 is used)
* BLOM uses potential temperature referred to the ocean surface (i.e., $\theta_0$), while WOA uses *in-situ* temperature.
* the potential temperature will on average is about $0.4^{\circ}$ colder than *in-situ* temperature in the abyssal part of the ocean.
* This inconsistency may give some artificial bias when comparing the model climatology to WOA climatology.

In this fix, we provide a fix to use derived potential temperature of WOA13.  It is also possible to still use *in-situ* WOA13 temperature to have backward compatibility.

Potential temperature is calculated based on WOA13 in-situ temperature (`t_an`), salinity (`s_an`) and depth.
* Reference level at ocean surface.
* Atmospheric pressure is ignored.
* NCL functions [potmp_insitu_ocn](https://www.ncl.ucar.edu/Document/Functions/Built-in/potmp_insitu_ocn.shtml), and [depth_to_pres](https://www.ncl.ucar.edu/Document/Functions/Built-in/depth_to_pres.shtml) are used.

This will affect the:
* Hovmouller plot
* annual climatology at depths
* zonal mean temperature
* equatorial section plots

Related to issue #25 

**In summary:**
* The NorESM diagnostic tool, `diag_run` now uses potential temperature of WOA13 as default. 
* To use *in-situ* temperature as normal, one should enter the *passive mode* (`diag_run .... -p`, see more in the [documentation](https://noresm-docs.readthedocs.io/en/noresm2/diagnostics/diag_run.html#passive-mode)) and uncomment `#export WOA13_TEMP_TYPE=in-situ`.

A test comparison of the results are shown below.

### CODE REVIEW

**Comment by YanchunHe** [comment]
# Difference of annual mean in-situ and potential temperature from WOA13
(in-situ minus potential temperature)

## @ 3000 m
potmp_ann_diff_3000m_t_an.pdf

## @ 4000 m
potmp_ann_diff_4000m_t_an.pdf


## global zonal mean
woa13_decav_potmp00_01_zm_glb_diff

**Comment by YanchunHe** [comment]
## Example of how does it effect the model-obs comparison

Use NorESM2-MM piControl, years 1450-1649 mean, as an example
| using in-situ temp | using potential temperature|
| --- | --- |
| global zonal mean using in-situ temp | global zonal mean using potential temp |
| annual mean at 3000m, in-situ temp| annual mean at 3000m, potential temperature|
| 3000m using in-situ temp | 3000m using potential temp |
| equatorial, in-situ| equatorial, potential temperature|
| |
| hovmoller, in-situ| hovmoller, potential temperature|
||  |
